yt/frontends/halo_catalog/tests/test_outputs.py

```diff
@@ -78,11 +86,46 @@ def test_halo_catalog_boundary_particles(self):
         fn = fake_halo_catalog(data)
         ds = yt_load(fn)
 
-        assert isinstance(ds, HaloCatalogDataset)
+        assert isinstance(ds, YTHaloCatalogDataset)
 
         for field in ["particle_mass"]:
             f1 = data[field].in_base()
             f1.sort()
             f2 = ds.r[field].in_base()
             f2.sort()
             assert_array_equal(f1, f2)
+
+
+t46 = "tiny_fof_halos/DD0046/DD0046.0.h5"
+
+
+@requires_file(t46)
+@requires_module("h5py")
+def test_halo_quantities():
+    ds = data_dir_load(t46)
+    ad = ds.all_data()
+    for i in range(ds.index.total_particles):
+        hid = int(ad["halos", "particle_identifier"][i])
+        halo = ds.halo("halos", hid)
+        for field in ["mass", "position", "velocity"]:
+            v1 = ad["halos", "particle_%s" % field][i]
+            v2 = getattr(halo, field)
+            assert_allclose_units(
```

Comment:
if I'm reading this right, you're the arrays you're comparing shouldn't just be approximately equal but rather be the same exact objects ?
so should this be
```python
assert v1 is v2
```
or
```python
assert_array_equal(
```
?